extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapStoreIntegrationTest.java

```diff
@@ -330,4 +329,30 @@ public void testDestroy() {
             assertDoesNotContainRow(client, "SHOW MAPPINGS", rows);
         }, 5);
     }
+
+    @Test
+    public void testInstanceShutdown() throws Exception {
+        HazelcastInstance client = client();
+        IMap<Integer, Person> map = client.getMap(tableName);
+
+        // create another member
+        HazelcastInstance hz3 = factory().newHazelcastInstance(memberConfig);
+        assertClusterSizeEventually(3, hz3);
+
+        // fill the map with some values so each member gets some items
+        for (int i = 1; i < 1000; i++) {
```

Comment:
```suggestion
        for (int i = 1; i <= 1000; i++) {
```

I am confused why tests passed, adding should stop at 999

Reply:
I tried (and failed) to explain this in the PR description:

> Note that the test passes without the new check in the destroy method. It is because when the node is shutting down the SQL query to remove the mapping can't be executed, so it's not removed.

Previous behaviour:

We insert items up to 999. We shut down the member. The MapStore destroy method is called. The drop mapping sql command is not executed because the instance is shutting down (exception in the log). -> it still exists on the other nodes -> it loads item 1000 from DB and tests pass.

Reply:
It does not matter if you populate the map with 10,100 or 1000 items or not.  Just terminate one HZ instance and then execute the "INSERT INTO "
If mapping is not destroyed it succeeds , otherwise it fails

Reply:
Yeah, I was confused that we are inserting 999 items, but expecting 1000 of them, but seems clear now, thanks

Reply:
It kinda matters - with 10 items it might happen that all will end up on a single member, and the other member will not run initialize the map store, making the test moot.